docker/catalogue/Dockerfile

```diff
@@ -1,14 +1,46 @@
 FROM golang:1.6
 
-RUN mkdir /app
-COPY . /go/src/github.com/microservices-demo/catalogue/
-COPY images/ /images/
+COPY . /go/src/github.com/microservices-demo/catalogue
+WORKDIR /go/src/github.com/microservices-demo/catalogue
 
 RUN go get -u github.com/FiloSottile/gvt
-RUN cd /go/src/github.com/microservices-demo/catalogue && gvt restore
+RUN gvt restore && \
+    CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /app github.com/microservices-demo/catalogue/cmd/cataloguesvc
+
+FROM alpine:3.4
+
+ENV	SERVICE_USER=myuser \
+	SERVICE_UID=10001 \
+	SERVICE_GROUP=mygroup \
+	SERVICE_GID=10001
+
+RUN	addgroup -g ${SERVICE_GID} ${SERVICE_GROUP} && \
+	adduser -g "${SERVICE_NAME} user" -D -H -G ${SERVICE_GROUP} -s /sbin/nologin -u ${SERVICE_UID} ${SERVICE_USER} && \
+	apk add --update libcap
```

Comment:
image can be slightly reduced bu removing apk cache: `apk add --no-cache libcap`, or add `rm -rf /var/cache/apk/*`
maybe we could consider using the scratch image as base image instead of alpine

Reply:
I'd recommend copying the LinuxKit pattern (see e.g. [`linuxkit/kubernetes/pkg/kube-e2e-test/Dockerfile`](https://github.com/linuxkit/kubernetes/blob/master/pkg/kube-e2e-test/Dockerfile), an one of [my adaptations of it](https://github.com/errordeveloper/seldon-gitops) without using LinuxKit's base image). But I wouldn't worry about this right now.